gammapy/datasets/map.py

```diff
@@ -2321,37 +2329,298 @@ def peek(self, figsize=(12, 8)):
 
         """
 
+        def make_rgb_transparent(rgba, bg_rgb=(1, 1, 1)):
+            """
+            Mix an RGBA color with a background RGB color based on its alpha value.
+
+            Parameters
+            ----------
+            rgba : iterable
+                A 4-element sequence representing the color in RGBA.
+            bg_rgb : tuple, optional
+                The background RGB color (default is white).
+
+            Returns
+            -------
+            list
+                The blended RGB color as [R, G, B].
+            """
+            rgb = rgba[:3]
+            alpha = rgba[3]
+            return [alpha * c1 + (1 - alpha) * c2 for c1, c2 in zip(rgb, bg_rgb)]
+
+        def create_custom_colormap(npred_min, cmap_name="viridis"):
+            """
+            Create two custom colormaps based on the given colormap and "GnBu".
+
+            This function:
+            - Resamples the specified colormap to 256 colors.
+            - Applies a power-law transform to the RGBA values for a custom transparency ramp.
+            - Sets any "bad" values in the colormap to the color representing `npred_min`.
```

Comment:
not sure to understand the logic here
bad values should not be in the range of the colormap, typically nan should appear in white or black

Reply:
deprecated, if using afmhot and correctly setting bad to black, there is no need to introduce this function.